Khan/wonder-blocks
# Migrate ModalLauncher documentation to storybook

## Summary:
Migrate ModalLauncher documentation to Storybook using
Styeguidist docs as a foundation.

Update OnePaneDialog with the "Flexible Modal" example from
Styleguidist.

NOTE: The JS Docs don't currently show up at the top of the
page because there is something wrong with the integration.
@jandrade is working on a fix.

Storybook publish: 
https://5e1bf4b385e3fb0020b7073c-eyfgklpjyu.chromatic.com/?path=/docs/modal-modallauncher--default

Issue: https://khanacademy.atlassian.net/browse/WB-1141

## Demo of the flexible modal in OnePaneDialog:

https://user-images.githubusercontent.com/13231763/186538331-213edb69-105e-47a8-b417-43f9cb80d0b1.mov



## Test plan:
- Open storybook and go to ModalLauncher
- Check that all the prop types look correct.
- Read through and make sure all the story descriptions make
  sense and that there are no typos.
- Interact with all the modals and make sure they behave as expected.
